src/components/UI/ToggleSwitch/index.css

```diff
@@ -1,82 +1,54 @@
-/* The switch - the box around the slider */
-.switch {
-  margin-inline-start: 13px;
-  margin-block-start: 1.25px;
-  position: relative;
-  display: inline-block;
-  width: 16px;
-  height: 16px;
+.toggleSwitchWrapper {
+  display: grid;
+  grid-template-areas: 'check text';
+  grid-template-columns: min-content 1fr;
+  align-items: center;
+  justify-items: flex-start;
+  cursor: pointer;
 }
 
-/* Hide default HTML checkbox */
-.switch input {
+.hiddenCheckbox {
   display: none;
-  width: 0;
-  height: 0;
-}
-
-.toggleWrapper span {
-  transition: 350ms;
-  color: var(--text-default);
 }
 
-.toggleWrapper:hover span {
+/* add fake checkbox */
+.toggleSwitchWrapper:before {
+  content: '';
+  display: inline-block;
+  margin-inline-start: 13px;
+  margin-inline-end: 0.5rem;
+  width: 20px;
+  height: 20px;
   transition: 350ms;
-  color: var(--accent);
-}
-
-.switch:hover .checkmark,
-.toggleWrapper:hover .switch .checkmark {
-  border-color: var(--accent-overlay);
-}
-
-.switch:hover input:checked ~ .checkmark,
-.toggleWrapper:hover .switch input:checked ~ .checkmark {
-  border-color: var(--accent-overlay);
-}
-
-.switch input:checked ~ .checkmark {
-  background-color: var(--accent);
+  border: 3px solid var(--accent);
+  border-radius: 2px;
+  grid-area: check;
+  color: var(--background);
+  background: var(--background);
+  font-size: 1.5rem;
+  line-height: 1.2rem;
+  font-weight: 900;
+  font-family: Arial, Helvetica, sans-serif;
 }
 
-.switch input:checked ~ .checkmark:after {
-  display: block;
+.toggleSwitchWrapper.checked:before {
+  content: '\2713';
```

Comment:
replaced the hacky check done using css borders and rotation with a real `check` character